packages/safe-kit/src/extensions/messages/offChainMessages.ts

```diff
@@ -0,0 +1,57 @@
+import { GetSafeMessageListProps, SafeMessageListResponse } from '@safe-global/api-kit'
+import { EIP712TypedData } from '@safe-global/safe-core-sdk-types'
+
+import { SafeClient } from '@safe-global/safe-kit/SafeClient'
+import { SafeMessageClient } from '@safe-global/safe-kit/extensions/messages/SafeMessageClient'
+import { SafeClientResult } from '@safe-global/safe-kit/types'
+
+/**
+ * Extend the SafeClient with the ability to use off-chain messages
+ *
+ * @example
+ * const safeClient = await createSafeClient({ ... })
+ *
+ * const safeMessagesClient = await safeClient.extend(
+ *   offChainMessages()
+ * )
+ *
+ * const { messages } = await safeMessagesClient.sendMessage(...)
+ * await safeMessagesClient.confirm(messages?.messageHash)
+ */
+export function offChainMessages() {
+  return (client: SafeClient) => {
+    const safeMessageClient = new SafeMessageClient(client.protocolKit, client.apiKit)
+
+    return {
+      /**
+       * Creates an off-chain message using the Transaction service
+       *
+       * @param {string | EIP712TypedData} message The message to be sent, can be a raw string or an EIP712TypedData object
+       * @returns {Promise<SafeClientResult>} A SafeClientResult. You can get the messageHash to confirmMessage() afterwards from the messages property
+       */
+      async sendMessage(message: string | EIP712TypedData): Promise<SafeClientResult> {
```

Comment:
Would it make sense to change the function name so it doesn't match the one from `onChainMessages()`? E.g. naming this function `sendOffChainMessage`? Otherwise sendMessage would be overwritten in case the SafeCient is extended by both, onChainMessages and offChainMessages.

Reply:
mmm, not sure, it's a long name :D. The thing is that the offchain messages are the default now. I don't know if this is better or override the send() method (instead creating the sendMessage) when the dev uses the onChainMessages extension

Reply:
at the end the onchain messages are regular transactions

Reply:
Changed